mkdocs/contrib/search/search_index.py

```diff
@@ -175,10 +170,17 @@ def __init__(self, *args, **kwargs) -> None:
         self.data: list[ContentSection] = []
         self.section: ContentSection | None = None
         self.is_header_tag = False
+        self.is_permalink = False
         self._stripped_html: list[str] = []
 
     def handle_starttag(self, tag: str, attrs: list[tuple[str, str | None]]) -> None:
         """Called at the start of every HTML tag."""
+        atts = dict(attrs)
+        # Check for permalink in header
+        if self.is_header_tag and tag == 'a' and 'headerlink' in (atts.get('class') or ''):
```

Comment:
'class' is a string. `'headerlink' in 'class1 adfgfheaderlinkdhsgh class2'` is true, that's probably not intended.

Reply:
Good call. We should definitely fix this.